### PR TITLE
fix: combine resourceowner query in reduce function for user grant

### PR DESCRIPTION
**Pull request by stebenz**
Restructure function to read resourceowners in a single query of different resources contained in the usergrant during the reduce function to fill the information into the projection. Additional test added for all combinations.

### Definition of Ready

- [X] I am happy with the code
- [X] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [X] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [X] No debug or dead code
- [X] My code has no repetitions
- [X] Critical parts are tested automatically
- [X] Where possible E2E tests are implemented
- [X] Documentation/examples are up-to-date
- [X] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

File: internal/query/projection/user_grant.go [hunk 1]
```diff
@@ -439,9 +442,8 @@ func getResourceOwners(ctx context.Context, es handler.EventStore, instanceID, u
 	if err != nil {
 		return "", "", "", err
 	}
-	if len(events) != eventCount {
```
Comment (livio-a) [inline internal/query/projection/user_grant.go:442]: eventCount can be removed and `reduceAddded` (in this file) does not yet check if the necessary resource owners are returned

**Review by livio-a — CHANGES_REQUESTED** [review]

File: internal/query/projection/user_grant.go (hunk 1 above)
Comment (stebenz) [reply internal/query/projection/user_grant.go:442]: will remove eventCount, but if all necessary resourceOwners are returned is checked in the places where this function is used as they are variable

Comment (livio-a) [reply internal/query/projection/user_grant.go:442]: https://github.com/zitadel/zitadel/pull/7383/files#diff-2e359312ea0b5419c077502f30b3ce835862614a1f11a399e1690ce8d60b6204R173-R176

Comment (stebenz) [reply internal/query/projection/user_grant.go:442]: If I try to open the link I just get this PRs file changes, what did you want to highlight?

**Comment by livio-a** [comment]
please provide a description and a meaningful title

File: internal/query/projection/user_grant.go (hunk 1 above)
Comment (stebenz) [reply internal/query/projection/user_grant.go:442]: totally missed that, added a function with error on missing resourceowner checks.

**Review by livio-a — APPROVED** [review]

**Pull request merged**